services/galley/src/Galley/API/Query.hs

```diff
@@ -90,14 +94,37 @@ import qualified Wire.API.Conversation.Role as Public
 import Wire.API.Error
 import Wire.API.Error.Galley
 import Wire.API.Federation.API
+import Wire.API.Federation.API.Brig
 import Wire.API.Federation.API.Galley
 import Wire.API.Federation.Client (FederatorClient)
 import Wire.API.Federation.Error
+import Wire.API.FederationStatus
 import qualified Wire.API.Provider.Bot as Public
 import qualified Wire.API.Routes.MultiTablePaging as Public
 import Wire.API.Team.Feature as Public hiding (setStatus)
 import Wire.Sem.Paging.Cassandra
 
+getFederationStatus :: Member FederatorAccess r => Local UserId -> RemoteDomains -> Sem r FederationStatus
+getFederationStatus _ req = do
+  firstConflictOrFullyConnected
+    <$> E.runFederatedConcurrently
+      (flip toRemoteUnsafe () <$> Set.toList req.rdDomains)
+      (\qds -> fedClient @'Brig @"get-federation-status" (DomainSet (tDomain qds `Set.delete` req.rdDomains)))
```

Comment:
```suggestion
      (\qds -> fedClient @'Brig @"get-federation-status" (DomainSet req.rdDomains))
```

it's both good, since we expect the target remote domain to also know about us.  but it's nice to make sure that they do :-).  and also the code looks arguably simpler now.